hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/MergeIntoHoodieTableCommand.scala

```diff
@@ -289,7 +290,7 @@ case class MergeIntoHoodieTableCommand(mergeInto: MergeIntoTable) extends Hoodie
     }
     var writeParams = parameters +
       (OPERATION.key -> operation) +
-      (HoodieWriteConfig.WRITE_SCHEMA.key -> getTableSchema.toString) +
+      (HoodieWriteConfig.WRITE_SCHEMA_OVERRIDE.key -> getTableSchema.toString) +
```

Comment:
These changes are necessary to align and streamline schema handling for `MERGE INTO` operation.

Changes:
 - Make sure it passes expected record's schema to the ExpressionPayload (via `PAYLOAD_RECORD_AVRO_SCHEMA` prop)
 - Make sure proper configs are specified to disable inapplicable operations such as: a) schema reconciliation, b) schema canonicalization (null/non-null), c) schema validation (since input to MI might not necessarily be matching the target table